additional test data: Blockquote, Callout, Embed, Text

srilq opened the pull request:
Changes:
* Adds new test data to `BlockquoteTest`, `CalloutTest`, `EmbedTest` and `TextTest`
* Modifies generator classes to satisfy new tests.

nigelhanlon commented on src/Text.php:
```diff
@@ -20,6 +20,8 @@ public static function beautify($markdown)
     {
         // remove whitespace on lines with no non-whitespace characters
         $markdown = preg_replace('/\n\s+\n/', "\n\n", $markdown);
+        $markdown = preg_replace('/^[^\S\n]+\n/', "\n", $markdown);
```
Given how many new lines we deal with, would it be worth looking at the `\R` character? I found an interesting article about it here: https://nikic.github.io/2011/12/10/PCRE-and-newlines.html. It may be over complicating things though as I am sure `\n` will work in 99% of cases. Feel free to ignore.

srilq replied:
@nigelhanlon How does this look?

```diff
diff --git a/src/Text.php b/src/Text.php
index 1e9ad06..4e677c6 100644
--- a/src/Text.php
+++ b/src/Text.php
@@ -9,6 +9,8 @@ class Text implements CopilotTag
     {
         if (!is_string($text)) throw new \InvalidArgumentException("Text::__construct first argument \$text must be a string. Given: ".($text ? "$text " : "")."(".gettype($text).").");
         $this->text = $text;
+        // convert newline types to LF newlines
+        $this->text = preg_replace('/\R/', "\n", $this->text);
     }
 
     public function write()
```

nigelhanlon replied:
lgtm 👍

srilq replied:
(see [Text.php line 13](https://github.com/conde-nast-international/copilot-markdown-generator-php/pull/7/files#diff-25cc76266925609cf768caf23952e0deR13))

amillward commented on src/Embed.php:
```diff
@@ -15,7 +15,8 @@ class Embed implements CopilotTag
     {
-        if (!is_string($uri)) throw new \InvalidArgumentException('Embed::__construct first argument $uri must be a string. Given:'.($uri?" $uri":'').' ('.gettype($uri).')');
+        if (!is_string($uri)) throw new \InvalidArgumentException("Embed::__construct first argument \$uri must be a string. Given: ".($uri ? "$uri " : "")."(".gettype($uri).").");
+        if (preg_match('/\s/', $uri)) throw new \InvalidArgumentException("Embed::__construct first argument \$uri cannot contain whitespace. Given: \"".str_replace("\n", "\\n", $uri)."\".");
```
What are the pros/cons of this? We could allow urls with leading/trailing whitespace and just trim them in the library?

srilq replied:
@amillward In tugboat, we removed any newlines and then trimmed leading/trailing spaces. Removing newlines and continuing silently feels wrong here, but allowing leading/trailing spaces seems fine, and it saves them trimming the input. I'll add this now.

srilq commented:
@nigelhanlon I've added the same functionality in `Link` as was added in `Embed` for consistency.

I've also added labels to tests! This will make things a bit easier. 😅
